### PR TITLE
Remove the dependency for HttpConfig from iocore/cache

### PR DESCRIPTION
**Pull request by maskit**

### CODE REVIEW

File: src/iocore/cache/CMakeLists.txt [hunk 1]
```diff
@@ -46,7 +46,7 @@ target_include_directories(inkcache PRIVATE ${CMAKE_SOURCE_DIR}/lib)
 
 target_link_libraries(
   inkcache
-  PUBLIC ts::aio ts::hdrs ts::http ts::inkevent ts::tscore
```
Comment (maskit) [inline src/iocore/cache/CMakeLists.txt:49]: All changes are to remove this `ts:http`.

**Comment by cmcfarlen** [comment]
[approve ci autest]

File: include/iocore/cache/CacheVC.h [hunk 2]
```diff
@@ -49,6 +49,8 @@
 #include <cstdint>
 
 struct Stripe;
+struct Vol;
```
Comment (cmcfarlen) [inline include/iocore/cache/CacheVC.h:52]: Is there still a `struct Vol`?

Comment (maskit) [reply include/iocore/cache/CacheVC.h:52]: I probably resolved conflict in a wrong way.

**Comment by maskit** [comment]
[approve ci Debian]
[approve ci Rocky]

**Comment by maskit** [comment]
[approve ci debian]

**Comment by maskit** [comment]
[approve ci rocky]

**Comment by maskit** [comment]
[approve ci debian]

**Review by bneradt — APPROVED** [review]
Very nice.

**Pull request merged**